database/changesets/20210422_TerraDataRepoSourceDetailsTable.xml

```diff
@@ -18,9 +18,12 @@
         <sql>
             CREATE TYPE TerraDataRepoSourceDetails AS (
                 id                          bigint,      -- unique to this table
+                start_time                  timestamptz, -- start timestamp of the TDR data query interval
+                end_time                    timestamptz, -- end timestamp of the TDR data query interval
+                datarepo_row_ids            text[],      -- array of datarepo row ids for snapshot creation
                 snapshot_creation_job_id    text,        -- id of the TDR snapshot creation job
-                snapshot_id                 text,        -- id of the snapshot lives in TDR
-                datarepo_row_id             text         -- id of the TDR data row
+                updated                     timestamptz, -- status update time
+                consumed                    boolean      -- whether or not this is consumed yet
```

Comment:
`-- true when consumed, false otherwise`
… but maybe …
`consumed timestamptz, -- time when consumed, null otherwise`
is better?

Reply:
Actually, I thought `updated` did the `consumed timestamptz, -- time when consumed, null otherwise` job when it's consumed combining with the `boolean`?

Reply:
But that logic is contained in one field which is a bit nicer :)

Reply:
cool, I will move forward and update then!